File: java/WordCount.java
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.rdd.RDD;
import scala.Tuple2;

import java.util.Arrays;

public class WordCount {
    public static void main(String[] args) {
        long start1 = System.currentTimeMillis();

        SparkConf conf = new SparkConf()
                .setAppName("Spark Vs Python")
                .setMaster("local[1]");
        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaRDD<String> text_file = sc.textFile("data/random_text.txt");
        JavaPairRDD<Object, Integer> count = text_file
                .flatMap(line -> Arrays.asList(line.split(" ")).iterator())
                .mapToPair(word -> new Tuple2<Object, Integer>(word, 1))
                .reduceByKey(Integer::sum);
        count.saveAsTextFile("src/output_java");


        long end1 = System.currentTimeMillis();
        System.out.println("Elapsed Time in mili seconds: "+ (end1-start1));

    }
}

File: scala/wordcount.scala
import org.apache.spark.rdd.RDD
import org.apache.spark.{SparkConf, SparkContext}

object wordcount {

  def main(args: Array[String]): Unit = {
    val start1: Long = System.currentTimeMillis

    val conf: SparkConf = new SparkConf()
    .setAppName("Spark Performance Word Count")
    .setMaster("local[1]")
    val sc = new SparkContext(conf)
    val text_file: RDD[String] = sc.textFile("data/random_text.txt")
    val count: RDD[(String, Int)] = text_file.flatMap(line => line.split(" "))
                                              .map(word  => (word, 1))
                                              .reduceByKey(_+_)
                                              .sortBy(_._2)

    count.saveAsTextFile("data/output_scala")

    val end1: Long = System.currentTimeMillis
    System.out.println("Elapsed Time in mili seconds: " + (end1 - start1))


  }
}